src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -114,23 +121,38 @@ static void HandleNodeIdResolve(void * context, DnssdService * result, CHIP_ERRO
     VerifyOrDie(proxy != nullptr);
 
     ResolvedNodeData nodeData;
-    Platform::CopyString(nodeData.mHostName, result->mHostName);
-    nodeData.mInterfaceId = result->mInterface;
-    nodeData.mAddress[0]  = result->mAddress.ValueOr({});
-    nodeData.mPort        = result->mPort;
-    nodeData.mNumIPs      = 1;
+    Platform::CopyString(nodeData.mHostName, firstResult->mHostName);
+    nodeData.mInterfaceId = firstResult->mInterface;
+    nodeData.mPort        = firstResult->mPort;
     nodeData.mPeerId      = peerId;
     // TODO: Use seconds?
     const System::Clock::Timestamp currentTime = System::SystemClock().GetMonotonicTimestamp();
 
-    nodeData.mExpiryTime = currentTime + System::Clock::Seconds16(result->mTtlSeconds);
+    nodeData.mExpiryTime = currentTime + System::Clock::Seconds16(firstResult->mTtlSeconds);
 
-    for (size_t i = 0; i < result->mTextEntrySize; ++i)
+    size_t addressesFound = 0;
+    for (auto & result : results)
```

Comment:
same comment as above here:
  - interface id seems wrong. we have to assert/check or split.
  - text entries may also be duplicated here. What is our strategy?

Reply:
Done by changing the API so this is a non-issue.